srunner/scenarios/parking_exit.py

```diff
@@ -0,0 +1,174 @@
+from srunner.scenariomanager.scenarioatomics.atomic_trigger_conditions import InTriggerDistanceToLocation
+from srunner.scenarios.basic_scenario import BasicScenario
+
+import math
+import operator
+
+
+def convert_dict_to_location(actor_dict):
+    """
+    Convert a JSON string to a Carla.Location
+    """
+    location = carla.Location(
+        x=float(actor_dict['x']),
+        y=float(actor_dict['y']),
+        z=float(actor_dict['z'])
+    )
+    return location
+
+
+def get_same_dir_transform_by_distance(transform, distance: float):
+    """
+    Get a transform in the same direction as the given one.
+    With {distance} (m) offset.
+    Works for arbitrary point on the map.
+    """
+    new_t = carla.Transform()
+    yaw = transform.rotation.yaw
+    new_t.location.x = transform.location.x + \
+        distance*math.cos(math.radians(yaw))
+    new_t.location.y = transform.location.y + \
+        distance*math.sin(math.radians(yaw))
+    new_t.location.z = transform.location.z
+    new_t.rotation = transform.rotation
+    return new_t
+
+
+class ParkingExit(BasicScenario):
+    """
+    This class holds everything required for a scenario in which the ego would be teleported to the parking lane.
+    Once the scenario is triggered, the OutsideRouteLanesTest will be deactivated since the ego is out of the driving lane.
+    Then blocking vehicles will be generated in front of and behind the parking point.
+    The ego need to exit from the parking lane and then merge into the driving lane.
+    After the ego is {end_distance} meters away from the parking point, the OutsideRouteLanesTest will be activated and the scenario ends.
+    
+    Note 1: For route mode, this shall be the first scenario of the route. The trigger point shall be the first point of the route waypoints.
+    
+    Note 2: Make sure there are enough space for spawning blocking vehicles.
+    """
+
+    def __init__(self, world, ego_vehicles, config, debug_mode=False, criteria_enable=True,
+                 timeout=180):
+        """
+        Setup all relevant parameters and create scenario
+        and instantiate scenario manager
+        """
+        self._world = world
+        self._map = CarlaDataProvider.get_map()
+        self.timeout = timeout
+        self._parking_point = convert_dict_to_location(
+            config.other_parameters['parking_point'])
+
+        if 'front_vehicle_distance' in config.other_parameters:
+            self._front_vehicle_distance = float(
+                config.other_parameters['front_vehicle_distance']['value'])
+        else:
+            self._front_vehicle_distance = 20  # m
+
+        if 'behind_vehicle_distance' in config.other_parameters:
+            self._behind_vehicle_distance = float(
+                config.other_parameters['behind_vehicle_distance']['value'])
+        else:
+            self._behind_vehicle_distance = 5  # m
+
+        if 'end_distance' in config.other_parameters:
```

Comment:
Again, not really needed, easily automatizable. We have the distance of the vehicle in front, so just use that distance plus some value

Reply:
end_distance parameter removed. Can be calculated by front_vehicle_distance

Reply:
end_distance parameter removed. Can be calculated by front_vehicle_distance